DogScepterLib/Core/GMWarning.cs

```diff
@@ -1,34 +1,70 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
+﻿namespace DogScepterLib.Core;
 
-namespace DogScepterLib.Core
+/// <summary>
+/// Class used for warnings that happen when dealing with <see cref="GMData"/>.
+/// </summary>
+public class GMWarning
 {
-    public class GMWarning
+    /// <summary>
+    /// A message that describes the current warning.
+    /// </summary>
+    public readonly string Message;
```

Comment:
Yeah... this is probably one of the least well-designed sections of the codebase. The whole warning system could potentially be refactored at some point, especially with the unused fields, or to consolidate the way they get logged. When I was initially writing this, I didn't know what I was going to be doing with the warnings in external code (in terms of logging, etc.).